src/libraries/System.Private.CoreLib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -62,6 +83,21 @@ internal virtual int LastIndexOf(T[] array, T value, int startIndex, int count)
 #endif
     }
 
+    internal sealed class DelegateEqualityComparer<T> : EqualityComparer<T>
```

Comment:
Why not to override `Equals(object)` and `GetHashCode()` to be able compare equality comparers?
Delegates are equality comparable, so it is possible to do.
Some parts of BCL uses comparing comparers to run more optimal code.

Reply:
Added.